openssl-tests/src/ffdhe_kx_with_openssl.rs

```diff
@@ -0,0 +1,231 @@
+use std::fs::{self, File};
+use std::io::{BufReader, Read, Write};
+use std::net::{TcpListener, TcpStream};
+use std::{str, sync::Arc, thread};
```

Comment:
nit: I think `sync::Arc` would be its own `use` line since it doesn't share a direct parent with `str` and `thread`.